doozer/doozerlib/build_info.py

```diff
@@ -166,24 +191,148 @@ def get_installed_package_build_dicts(self) -> Dict[str, Dict]:
+        # It should start with sha256: for now. Let's raise an error if this changes.
+        if not digest.startswith("sha256:"):
             raise ValueError(f"Received unrecognized digest {digest} for archive {self.get_archive_id()}")
         return digest
 
+    def get_manifest_list_digest(self):
+        return self.build_record_inspector.get_manifest_list_digest()
+
+
+class KonfluxImageInspector(ImageInspector):
+    def __init__(self, runtime: "doozerlib.Runtime", image_info: Dict,
+                 build_record_inspector: 'KonfluxBuildRecordInspector' = None):
+        super().__init__(runtime, build_record_inspector)
+        self._image_info = Model(image_info)
+        assert self._image_info['name'] == build_record_inspector.get_build_obj().image_pullspec
+
+    def get_build_inspector(self):
+        return self.build_record_inspector
+
+    def get_digest(self):
+        return self._image_info.digest
+
+    def get_image_meta(self):
+        return self.build_record_inspector.get_image_meta()
+
+    def get_pullspec(self):
+        return self._image_info['name']
+
+    def image_arch(self):
+        return brew_arch_for_go_arch(self._image_info.config.architecture)
+
+    def get_manifest_list_digest(self):
+        return self._image_info.listDigest
+
+
+class BuildRecordInspector(ABC):
+    def __init__(self, runtime):
+        self.runtime = runtime
+        self._nvr: Optional[str] = None  # Will be resolved to the NVR for the image/manifest list
+        self._build_pullspec = None  # Will track the pullspec of the image/manifest list
+        self._cache = dict()
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}:{self.get_build_id()}:{self.get_nvr()}'
+
+    @classmethod
+    def get_build_record_inspector_cls(cls, runtime):
+        build_system = runtime.build_system
+
+        if build_system == 'brew':
+            return BrewBuildRecordInspector
+
+        elif build_system == 'konflux':
+            return KonfluxBuildRecordInspector
+
+        else:
+            raise ValueError(f'Invalid build system: {build_system}')
+
+    @abstractmethod
+    def get_build_id(self):
+        pass
+
+    @abstractmethod
+    def get_build_obj(self):
+        pass
+
+    def get_nvr(self) -> str:
+        return self._nvr
+
+    @abstractmethod
+    def get_all_installed_package_build_dicts(self):
+        pass
+
+    @abstractmethod
+    def get_rpms_in_pkg_build(self, build_id: int):
+        pass
+
+    @abstractmethod
+    def get_image_meta(self):
+        pass
+
+    @staticmethod
+    def get_source_git_commit():
+        pass
+
+    @staticmethod
+    def find_non_latest_rpms():
+        pass
+
+    @abstractmethod
+    def get_version(self):
+        pass
+
+    @abstractmethod
+    def get_release(self):
+        pass
+
+    @abstractmethod
+    def get_build_webpage_url(self):
+        pass
 
-class BrewBuildImageInspector:
+    @abstractmethod
+    def is_under_embargo(self):
```

Comment:
`is_embargoed` would be cleaner imo

Reply:
This comes from the old code. I've tried not to change unnecessary names to keep changes at a minimum, otherwise the diff would have become truly unreadable. We can address this as a follow up